packages/data-stores/src/queries/use-launchpad.ts

```diff
@@ -0,0 +1,98 @@
+import apiFetch from '@wordpress/api-fetch';
+import { useQuery } from 'react-query';
+import wpcomRequest, { canAccessWpcomApis } from 'wpcom-proxy-request';
+
+interface APIFetchOptions {
+	global: boolean;
+	path: string;
+}
+
+interface Task {
+	id?: string;
+	completed?: boolean;
+	disabled?: boolean;
+	title?: string;
+	subtitle?: string;
+	badgeText?: string;
+	actionDispatch?: () => void;
+	isLaunchTask?: boolean;
+	warning?: boolean;
+}
+
+interface ChecklistStatuses {
+	links_edited?: boolean;
+	site_edited?: boolean;
+	site_launched?: boolean;
+	first_post_published?: boolean;
+	video_uploaded?: boolean;
+	publish_first_course?: boolean;
+	plan_completed?: boolean;
+	domain_upsell_deferred?: boolean;
+}
+
+interface LaunchpadResponse {
+	site_intent?: string | null;
+	launchpad_screen?: string | boolean | null | undefined;
+	checklist?: Task[] | null;
+	checklist_statuses?: ChecklistStatuses;
+}
```

Comment:
I added more rigorous typing for the response from the endpoint. There was a TS error on this, and we could have resolved it by returning any. This is what we're effectively doing in the original useLaunchpad hook - returning any.